compiler/test/unit/frontend.d

```diff
@@ -388,5 +388,5 @@ unittest
     t.module_.fullSemantic();
 
     assert(global.errors == 1);
-    assert(diagnosticMessages[0] == "is a Ddoc file, cannot import it");
+    assert(diagnosticMessages[$ - 32 .. $] == "is a Ddoc file, cannot import it");
```

Comment:
You want `diagnosticMessages[0][$ - 32 .. $]` there. Also I'd do it as 
```d
string msg = "is a Ddoc file, cannot import it";
assert(diagnosticMessages[0][$ - msg.length] == msg);
```
as it is more clear.

Reply:
You're right.